### PR TITLE
New `TextLog` archetype

### PR DESCRIPTION
**Pull request by emilk**
### What
* Part of https://github.com/rerun-io/rerun/issues/2793

Adds
* component `TextLogLevel`
* archetype `TextLog`

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)
* [x] I've included a screenshot or gif (if applicable)
* [x] I have tested [demo.rerun.io](https://demo.rerun.io/pr/3261) (if applicable)

- [PR Build Summary](https://build.rerun.io/pr/3261)
- [Docs preview](https://rerun.io/preview/6b7d5facdeb2da0d3a04b52788e5ac42e472ae7b/docs) 
- [Examples preview](https://rerun.io/preview/6b7d5facdeb2da0d3a04b52788e5ac42e472ae7b/examples) 
- [Recent benchmark results](https://ref.rerun.io/dev/bench/)
- [Wasm size tracking](https://ref.rerun.io/dev/sizes/)

### CODE REVIEW

File: docs/content/reference/data_types.md [hunk 1]
```diff
@@ -39,5 +39,4 @@ generic container for arrays of data. Images are restricted to tensors of rank 2
 
 ## Other **Archetypes**
 * [Scalar](data_types/scalar.md): a single scalar / metric value. Can be viewed in the `TimeSeries` space view.
-* [TextEntry](data_types/text_entry.md): captures text data. Can be viewed in the `Text` space view.
```
Comment (Wumpf) [inline docs/content/reference/data_types.md:42]: bit unfortunate to throw it away entirely and thus any mention of the ability to log text.
But I suppose putting up the new doc pages with all the examples and generated images is orthogonal

File: rerun_cpp/src/rerun/components/class_id.cpp [hunk 2]
```diff
@@ -1,4 +1,4 @@
-// DO NOT EDIT!: This file was auto-generated by crates/re_types_builder/src/codegen/cpp/mod.rs:54.
```
Comment (Wumpf) [inline rerun_cpp/src/rerun/components/class_id.cpp:1]: offtopic: The line number in here is really silly an causes lots of unnecessary diff

File: examples/python/signed_distance_fields/main.py [hunk 3]
```diff
@@ -79,21 +79,21 @@ def read_mesh(path: Path) -> Trimesh:
     return cast(Trimesh, mesh)
 
 
-@log_timing_decorator("global/voxel_sdf", rr.LogLevel.DEBUG)  # type: ignore[misc]
+@log_timing_decorator("global/voxel_sdf", "DEBUG")  # type: ignore[misc]
```
Comment (Wumpf) [inline examples/python/signed_distance_fields/main.py:82]: here and in all other python example code: Why not use constants for the level strings?

**Review by Wumpf — APPROVED** [review]
lgtm but I don't understand the change in python to string literals everywhere - there is documentation and attributes set on `TextLogLevelType` but then they aren't used anywhere?

File: examples/python/signed_distance_fields/main.py (hunk 3 above)
Comment (emilk) [reply examples/python/signed_distance_fields/main.py:82]: Truth? Because I am fed up with Python and so I threw my hands in the air and said "what's the difference - the errors are only caught at runtime anyway - at least a string literal is short"

File: rerun_cpp/src/rerun/components/class_id.cpp (hunk 2 above)
Comment (emilk) [reply rerun_cpp/src/rerun/components/class_id.cpp:1]: fixed

File: examples/python/signed_distance_fields/main.py (hunk 3 above)
Comment (Wumpf) [reply examples/python/signed_distance_fields/main.py:82]: 😢

File: crates/re_data_ui/src/component_ui_registry.rs [hunk 4]
```diff
@@ -83,44 +82,6 @@ fn format_arrow(value: &dyn arrow2::array::Array) -> String {
 
 // ----------------------------------------------------------------------------
 
-impl DataUi for re_components::TextEntry {
```
[the remark this answers is not in the document]
Comment (emilk) [reply crates/re_data_ui/src/component_ui_registry.rs:86]: The new components are just plain text, and yes, they show up nicely!

The old `TextEntry` had both text and level bundled together, which is why it required its own ui

File: examples/python/signed_distance_fields/main.py (hunk 3 above)
Comment (emilk) [reply examples/python/signed_distance_fields/main.py:82]: …also I thought I hade opened this PR in draft mode 😅

Comment (emilk) [reply examples/python/signed_distance_fields/main.py:82]: I also don't know how to do it. `rr2.components.TextLogLevelType.INFO` doesn't work: `module 'rerun.components' has no attribute 'TextLogLevelType'`

Comment (emilk) [reply examples/python/signed_distance_fields/main.py:82]: I'll fix it in a later PR
